echo.go

```diff
@@ -643,32 +646,48 @@ func (e *Echo) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 
 // Start starts an HTTP server.
 func (e *Echo) Start(address string) error {
+	e.startupMu.Lock()
 	e.Server.Addr = address
-	return e.StartServer(e.Server)
+	if err := e.configureServer(e.Server); err != nil {
+		e.startupMu.Unlock()
```

Comment:
As these function are not called very often we can use `defer` to improve readability.

Reply:
plain defer will not work to unlock mutex because function ends with blocking `return e.serve()` until server is shutdown and intention is to lock only when we are setting up listener and server.

Reply:
Ah yes!